docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Ranking/FastTree.cs

```diff
@@ -0,0 +1,104 @@
+
+namespace Samples.Dynamic.Trainers.Ranking
+{
+    public static class FastTree
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Create a list of training data points.
+            var dataPoints = GenerateRandomDataPoints(1000);
+
+            // Convert the list of data points to an IDataView object, which is consumable by ML.NET API.
+            var trainingData = mlContext.Data.LoadFromEnumerable(dataPoints);
+
+            // Define the trainer.
+            var pipeline = mlContext.Ranking.Trainers.FastTree();
+
+            // Train the model.
+            var model = pipeline.Fit(trainingData);
+
+            // Create testing data. Use different random seed to make it different from training data.
+            var testData = mlContext.Data.LoadFromEnumerable(GenerateRandomDataPoints(500, seed:123));
+
+            // Run the model on test data set.
+            var transformedTestData = model.Transform(testData);
+
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
+
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(5))
+                Console.WriteLine($"Label: {p.Label}, Score: {p.Score}");
+
+            // Expected output:
+			//   Label: 5, Score: 9.94396
+			//   Label: 4, Score: 2.556023
+			//   Label: 4, Score: -5.264654
+			//   Label: 1, Score: -8.955013
+			//   Label: 1, Score: -3.514952
+
+            // Evaluate the overall metrics
+            var metrics = mlContext.Ranking.Evaluate(transformedTestData);
+            PrintMetrics(metrics);
+            
+            // Expected output:
+			//   DCG: @1:23.80, @2:37.17, @3:46.52
+			//   NDCG: @1:0.56, @2:0.57, @3:0.60
+        }
+
+        private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count, int seed = 0, int groupSize = 10)
+        {
+            var random = new Random(seed);
+            float randomFloat() => (float)random.NextDouble();
+            for (int i = 0; i < count; i++)
+            {
+                var label = random.Next(0, 5);
+                yield return new DataPoint
+                {
+                    Label = (uint)label,
+                    GroupId = (uint)(i / groupSize),
+                    // Create random features that are correlated with the label.
+                    // For data points with larger labels, the feature values are slightly increased by adding a constant.
+                    Features = Enumerable.Repeat(randomFloat() + label * 0.1f, 50).ToArray()
```

Comment:
>Repeat(randomFloat() + label * 0.1f, 50).ToArray() [](start = 42, length = 50)

is this correct? I think it repeats the same number 50 times which is not what you want.

in other samples we do it like this:
Enumerable.Repeat(label, 50).Select(x => x + randomFloat()).ToArray()